src/parser/scanner_support.c

```diff
@@ -145,23 +219,85 @@ set_plan_include_hint (bool is_include)
   plan_include_hint = is_include;
 }
 
-void
-print_hint_dump (FILE * output)
+static void
+print_hint_stmt (FILE * output, int stmt_idx)
 {
   if (s_hint_msg.msg_ptr && *s_hint_msg.msg_ptr)
     {
-      fputs ("\nQuery hints:\n", output);
+      STMT_HINT *hint_ptr;
+
+      for (hint_ptr = &s_hint_msg.m_hint_head; hint_ptr; hint_ptr = hint_ptr->next)
+	{
+	  if (hint_ptr->stmt_no == stmt_idx)
+	    {
+	      break;
+	    }
+	}
 
-      s_hint_msg.msg_ptr[s_hint_msg.m_used] = '\0';
-      fprintf (output, "%s", s_hint_msg.msg_ptr);
+      if (hint_ptr)
+	{
+	  fputs ("\nQuery hints:\n", output);
+
+	  s_hint_msg.msg_ptr[s_hint_msg.m_used] = '\0';
+	  fprintf (output, "%s", s_hint_msg.msg_ptr + hint_ptr->offset);
+	  hint_ptr->stmt_no = -2;	/* change to no more use */
```

Comment:
what does mean "-2" ?
if it needs some meaning, use a macro defines a special meaning.

Reply:
Setting it to -2 (negative number) means the information is no longer used. I'll use define to make it clearer.